smoothscroll.js

```diff
@@ -1,156 +1,237 @@
-  var originalScrollBy = window.scrollBy;
-  var originalScrollIntoView = Element.prototype.scrollIntoView;
-
-  // store generally accessible frame id in case a new scroll animation is triggered before the previous
-  // completes, we can cancel the previous scroll.
-  var frame;
-
-  var startY, startX, endX, endY;
-
+  /*
+   * alias
+   * w: window global object
+   * doc: document
+   * undefined: undefined
+   */
+
+  // return if scrollBehavior is supported
+  if ('scrollBehavior' in doc.documentElement.style) return;
+
+  var SCROLL_TIME = 400,
+      // legacy scrolling methods
+      originalScrollTo = w.scrollTo,
+      originalScrollBy = w.scrollBy,
+      originalScrollIntoView = w.Element.prototype.scrollIntoView,
+      // global frame variable to avoid collision
+      frame,
+      // global metric variables
+      startY, startX, endX, endY;
+
+  /*
+   * returns actual time
+   * @method now
+   * @returns {Date}
+   */
   function now() {
-    return window.performance !== undefined && window.performance.now !== undefined ? window.performance.now() : Date.now !== undefined ? Date.now() : new Date().getTime();
+    return w.performance !== undefined && w.performance.now !== undefined ?
+      // if performance object supported return now, if not fallback to date object
+      w.performance.now() : Date.now !== undefined ? Date.now() : new Date().getTime();
   }
 
-  // ease-in-out
+  /*
+   * returns result of applying ease math function to a number
+   * @method ease
+   * @param {Number} k
+   * @returns {Number}
+   */
   function ease(k) {
     return 0.5 * (1 - Math.cos(Math.PI * k));
   }
 
-  function shouldBailOut(x) {
-    return typeof x === 'undefined' || x.behavior !== 'smooth';
+  /*
+   * returns true if first argument is an options object and contains a smooth behavior
+   * @method bailOut
+   * @param {Number|Object} x
+   * @returns {Boolean}
+   */
+  function bailOut(x) {
+    if (typeof x !== 'object' || x.behavior === undefined || x.behavior === 'auto' ) {
+      // first arg not an object, or behavior is auto or undefined
+      return true;
+    } else if (x.behavior === 'smooth') {
+      // first argument is an object and behavior is smooth
+      return false;
+    } else {
+      // behavior not supported
+      throw new TypeError(x.behavior + ' is not a valid value for enumeration ScrollBehavior');
+    }
+  }
+
+  /*
+   * changes scroll position inside an element
+   * @method scrollElement
+   * @params {Node} el
+   * @params {Number} x
+   * @params {Number} y
+   */
+  function scrollElement(el, x, y) {
+    el.scrollTop = y;
+    el.scrollLeft = x;
+  }
+
+  /*
+   * finds scrolalble parent of an element
+   * @method findScrollableParent
+   * @params {Node} el
+   */
+  function findScrollableParent(el) {
+    if (el.clientHeight < el.scrollHeight ||
+        el.clientWidth < el.scrollWidth) {
+      return el;
+    }
+
+    // only continue scaling if parentNode is valid
```

Comment:
I say @ebidel issue. Seriously I don't know that either. It could be a fix in a future version if you agree.